### PR TITLE
atomic action

### PR DESCRIPTION
**Pull request by matafela**
# Description
- Implement atomic action.
- Fix qpos seed shape error in MotionGenerator.
- Fix waypoint buliding error in ToppraPlanner
- example: ‵scripts/tutorials/sim/atom_action.py`

TODO:
- [x] Place and Move action
- [x] comments and docs
- [x] unittest

## Fixes # (issue)

- Fix qpos seed shape error in MotionGenerator.
- Fix waypoint buliding error in ToppraPlanner

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Enhancement (non-breaking change which improves an existing functionality)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the `black .` command to format the code base.
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Dependencies have been updated, if applicable.

### CODE REVIEW

File: scripts/tutorials/sim/atom_action.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+
+"""
+This script demonstrates the creation and simulation of a robot with a soft object,
```
Comment (yuecideng) [inline scripts/tutorials/sim/atom_action.py]: modify the description

File: embodichain/lab/sim/atomic_actions/__init__.py [hunk 2]
```diff
@@ -23,17 +23,18 @@
...
     ObjectSemantics,
     ActionCfg,
     AtomicAction,
 )
 from .actions import (
-    ReachAction,
-    GraspAction,
-    ReleaseAction,
     MoveAction,
+    PickUpAction,
```
Comment (yuecideng) [inline embodichain/lab/sim/atomic_actions/__init__.py:33]: Add docs for Atomic action, including the concepts, the AtomicActionEngine, and how to implement a atomic action,

File: embodichain/lab/sim/atomic_actions/actions.py [hunk 3]
```diff
@@ -22,523 +22,615 @@
...
-            ),
+        super().__init__(
+            motion_generator, cfg=cfg if cfg is not None else MoveActionCfg()
         )
 
-        result = self.plan_trajectory(target_states, options)
-
-        # Return PlanResult directly
-        return result
+        self.n_envs = self.robot.get_qpos().shape[0]
```
Comment (yuecideng) [inline embodichain/lab/sim/atomic_actions/actions.py:61]: Use `num_envs` for all batch simulation for the project

**Review by yuecideng — CHANGES_REQUESTED** [review]

File: scripts/tutorials/sim/atom_action.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    sim: SimulationManager = initialize_simulation(args)
+    robot = create_robot(sim)
+    mug = create_mug(sim)
+
+    motion_gen = MotionGenerator(
+        cfg=MotionGenCfg(planner_cfg=ToppraPlannerCfg(robot_uid=robot.uid))
+    )
+
+    pickup_cfg = PickUpActionCfg(
+        hand_open_qpos=torch.tensor(
```
Comment (yuecideng) [inline scripts/tutorials/sim/atom_action.py:194]: We should support joints that removed mimic
